app/src/main/java/org/mozilla/focus/search/ManualAddSearchEnginePreference.kt

```diff
@@ -43,29 +47,40 @@ class ManualAddSearchEnginePreference(context: Context, attrs: AttributeSet) :
 
         engineNameEditText = holder.findViewById(R.id.edit_engine_name) as EditText
         engineNameEditText?.addTextChangedListener(
-            buildTextWatcherForErrorLayout(
-                engineNameErrorLayout!!
-            )
+                buildTextWatcherForErrorLayout(
+                        engineNameErrorLayout!!
+                )
         )
         searchQueryEditText = holder.findViewById(R.id.edit_search_string) as EditText
         searchQueryEditText?.addTextChangedListener(
-            buildTextWatcherForErrorLayout(
-                searchQueryErrorLayout!!
-            )
+                buildTextWatcherForErrorLayout(
+                        searchQueryErrorLayout!!
+                )
         )
 
         progressView = holder.findViewById(R.id.progress) as ProgressBar
+        engineNameEditText?.setOnTouchListener { _, _ ->
+            lastEditTextTouched = SEARCH_ENGINE_NAME_KEY
+            false
+        }
+        searchQueryEditText?.setOnTouchListener { _, _ ->
+            lastEditTextTouched = SEARCH_QUERY_KEY
+            false
+        }
```

Comment:
Instead of using a field and touch handlers, could we use `isFocused()` in `onSaveInstanceState` to figure out which field is focused?

Reply:
@gouri-panda Do you think that's possible?

Reply:
Yes , I did that too.

Reply:
@pocmo Is there something I forgot to add :)

Reply:
It's not in this patch though? What I was wondering was whether we could remove things like `lastEditTextTouched` and the touch handlers and instead lookup which view is focused once the state needs to be saved. This has two advantages:
* Less code / state to keep
* It also works if you navigate via other means than touching (e.g. a keyboard)

Reply:
@pocmo Here i checked which edit text is focused on `onSaveInstanceState` and restores in `onRestoreInstanceState`. I am not sure which state are you talking about. Can you give an example with code? Thanks ;)